server/implementation/src/main/java/io/smallrye/graphql/execution/error/ExecutionErrorsService.java

```diff
@@ -84,10 +87,20 @@ private JsonObject toJsonError(GraphQLError error) {
 
         addKeyValue(objectBuilder, EXCEPTION, exception.getClass().getName());
         addKeyValue(objectBuilder, CLASSIFICATION, error.getErrorType().toString());
+        Map<String, Object> extensions = error.getExtensions();
+        populateCustomExtensions(objectBuilder, extensions);
 
         return Optional.of(objectBuilder.build());
     }
 
+    private void populateCustomExtensions(JsonObjectBuilder objectBuilder, Map<String, Object> extensions) {
+        if (Objects.nonNull(extensions)) {
```

Comment:
```suggestion
        if (extensions != null) {
```

`Objects.nonNull` is meant to be used for Stream-Predicates.